Research: Message ontology draft

jravenel opened the pull request:
@giacomodecolle here is the first draft if you are ok to check it out. 

cc @jupyter-naas/core-team

giacomodecolle commented on src/core/modules/ontology/ontologies/domain-level/MessageOntology.ttl:
```diff
@@ -0,0 +1,282 @@
+                                                   dc11:identifier "011-MSG" ;
+                                                   rdfs:label "message content"@en ;
+                                                   skos:definition "b is message content =Def b is a generically dependent continuant that is the information content entity constituting the semantic payload of a message"@en ;
+                                                   skos:example "The textual content of an email; the semantic content of a spoken message; the informational payload of a data packet"@en .
```
As they are defined, this might be identical with the message class above

giacomodecolle commented on src/core/modules/ontology/ontologies/domain-level/MessageOntology.ttl:
```diff
@@ -0,0 +1,282 @@
+                                                                              owl:onProperty <http://ontology.naas.ai/messaging#realizes> ;
+                                                                              owl:someValuesFrom <http://ontology.naas.ai/messaging#MessageReceiverRole>
+                                                                            ] ,
+                                                                            [ rdf:type owl:Restriction ;
```
wait is the process receiving the message? Shouldn't it be one of the agents in the process who receives it?

giacomodecolle commented on src/core/modules/ontology/ontologies/domain-level/MessageOntology.ttl:
```diff
@@ -0,0 +1,282 @@
+<http://ontology.naas.ai/messaging#MessageTransmissionProcess> rdf:type owl:Class ;
+                                                               rdfs:subClassOf bfo:0000015 ,
+                                                                               [ rdf:type owl:Restriction ;
+                                                                                 owl:onProperty <http://ontology.naas.ai/messaging#realizes> ;
```
these relations should be using the corresponding bfo/cco identifiers

giacomodecolle commented on src/core/modules/ontology/ontologies/domain-level/MessageOntology.ttl:
```diff
@@ -0,0 +1,282 @@
+                                                      rdfs:subClassOf bfo:0000023 ;
+                                                      dc11:identifier "014-MSG" ;
+                                                      rdfs:label "message sender role"@en ;
+                                                      skos:definition "b is a message sender role =Def b is a role that inheres in an agent & b is realized by processes in which that agent transmits information content to other agents"@en ;
```
This is good but fyi, it is kinda opaque what "sending information" means. I think it involves making a copy of some information, but what does "transmitting" means? That the copy is in one other location than the original one?

giacomodecolle commented on src/core/modules/ontology/ontologies/domain-level/MessageOntology.ttl:
```diff
@@ -0,0 +1,282 @@
+                                               rdfs:range <http://ontology.naas.ai/messaging#MessageContent> ;
+                                               dc11:identifier "001-MSG" ;
+                                               rdfs:label "has content"@en ;
+                                               skos:definition "b has content c =Def b is a message & c is message content & c is the information content entity that b bears"@en ;
```
see my comment below on message and message content

giacomodecolle commented on src/core/modules/ontology/ontologies/domain-level/MessageOntology.ttl:
```diff
@@ -0,0 +1,282 @@
+
+
+###  http://ontology.naas.ai/messaging#hasParticipant
+<http://ontology.naas.ai/messaging#hasParticipant> rdf:type owl:ObjectProperty ;
```
isn't this just the bfo relation?

giacomodecolle commented on src/core/modules/ontology/ontologies/domain-level/MessageOntology.ttl:
```diff
@@ -0,0 +1,282 @@
+                                             rdfs:range <http://ontology.naas.ai/messaging#Message> ;
+                                             dc11:identifier "004-MSG" ;
+                                             rdfs:label "receives"@en ;
+                                             skos:definition "p receives m =Def p is a message reception process & m is a message & m is the message that p obtains and processes"@en ;
```
why is the process receiving the message? I would say the agent with the receiver role is receiving the message

giacomodecolle commented on src/core/modules/ontology/ontologies/domain-level/MessageOntology.ttl:
```diff
@@ -0,0 +1,282 @@
+                                                dc11:identifier "006-MSG" ;
+                                                rdfs:label "triggered by"@en ;
+                                                skos:definition "p triggered by r =Def p is a response generation process & r is a message reception process & r is the causal antecedent that initiates p"@en ;
+                                                skos:example "A response generation process triggered by receiving an email; an automated reply process triggered by message reception"@en .
```
do we want to have this just for automated replies?

giacomodecolle commented on src/core/modules/ontology/ontologies/domain-level/MessageOntology.ttl:
```diff
@@ -0,0 +1,282 @@
+                                                            ] ;
+                                            dc11:identifier "010-MSG" ;
+                                            rdfs:label "message"@en ;
+                                            skos:definition "b is a message =Def b is a generically dependent continuant that bears information content intended for transmission between agents"@en ;
```
What about "a GDC that is concretized by at least two copies, and one of them is generated by an agent with the intention of having at least another agent notice the GDC"?

giacomodecolle reviewed (dismissed):
A good start, left comments inline. I think we can do something more precise with "transmission", hopefully soon I can share something with you from the CCO working group

jravenel commented:
@giacomodecolle, here a summary of the changes made following your  comments:

1) Fixed semantic vagueness around "transmitting information"
- Replaced opaque language with precise GDC concretization mechanics
- Transmission now defined as "creating new concretizations of message GDC in new locations/media"
- Reception defined as "accessing existing concretizations"

2) Clarified agent-process-role realization confusion
- Maintained BFO standard: processes realize roles that inhere in participant agents
- Updated definitions to emphasize roles inhere in agents, not processes

3) Fixed GDC vs ICE category error
- Eliminated MessageContent class (category error: GDCs ARE content, don't "bear" it)
- Added intentionality requirement to address random string problem
- Proper GDC concretization modeling with minimum 2 copies constraint

4) Removed domain-specific triggeredBy relation
- Replaced with BFO temporal modeling using precedes
- Eliminated scope limitation to automated systems

5) Fixed agent vs process semantic confusion
- Agents receive/transmit messages, processes have message input/output
- Proper BFO participant modeling with complex domain restrictions

6) Eliminated redundant BFO subproperties
- Removed custom hasParticipant and realizes (use BFO relations directly)
- Zero semantic redundancy, improved BFO compliance

I'll merge to clean the plate, and suggest you open a new PR if any additional comment, also because this will make you an official contributor on the github project page (only people that submit PR that get merge are added).
